### PR TITLE
pygments syntax: improve cylc lexer

### PR DESCRIPTION
**Pull request by oliver-sanders**
Support additional features of the Cylc suite.rc language to permit use in the Cylc documentation (cylc/cylc#2910).

Adds support for the following (see table in cylc/cylc#2752):
* Inter-Suite
* Include files
* xtriggers
* empy

Fixes:
* Line continuation characters in headings
   ```
    [foo, \
    bar]
    ```
* Cycle points in inter-cycle offsets

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Automated testing for syntax files, hmmm, that might be a tricky one.

The simplest thing I can think of doing would be to come up with some devilishly convoluted examples to run through each editor in turn, then perform a visual diff on before and after screenshots.

**Comment by oliver-sanders** [comment]
> If there is a way to test by giving the lexical analyser some suite.rc fragments and see what tokens come out on the other side

For `pygments`, *yes*.
For `vim`, *possibly*.
For `gedit` *god knows*.

See cylc/cylc#2915 for a discussion on testing approaches for syntax files.

**Comment by oliver-sanders** [comment]
Have added standard reference files for testing lexers in https://github.com/cylc/cylc/pull/2916.

PS/ Spotted an edge case with `empy` (`@[if []]`).

File: sphinx/ext/cylc_lang.py [hunk 1]
```diff
@@ -118,24 +127,37 @@ class CylcLexer(RegexLexer):
...
             include('parameterisation'),
+            (r'@\w+', GRAPH_XTRIGGER_TOKEN),
             (r'\w+', GRAPH_TASK_TOKEN),
+            (r'\!\w+', Other),
             (r'\s', Text),
             (r'=>', Operator),
-            (r'[\&\|\!]', Operator),
+            (r'[\&\|]', Operator),
             (r'[\(\)]', Punctuation),
             (r'\[', Text, 'intercycle-offset'),
```
Comment (sadielbartholomew) [inline sphinx/ext/cylc_lang.py]: I have noticed one omission. Dependencies on specific cycle points e.g. ``graph = "red[2019] => blue"`` are supported as valid specification within square brackets in a graph string, where only intercycle offsets appear to be recognised by the lexer as-is.

Comment (sadielbartholomew) [reply sphinx/ext/cylc_lang.py]: The good news is the lexer is now so water-tight it picked up on several missing markers in some of the Cylc documentation code excerpts. It's nearly there :clap:

**Comment by oliver-sanders** [comment]
>  cycle point arithmetic has not been included

Arithmetic is fine but arbitrary whitespace is not supported.

**Comment by sadielbartholomew** [comment]
> Arithmetic is fine but arbitrary whitespace is not supported.

Oh, fair enough, I simply used the example you provided in your original comment, assuming it was valid. Good!

**Pull request merged**